### PR TITLE
Make app data folder consistent per URL and allow clearing of app data from menu

### PR DESCRIPTION
**Pull request by garymoon**

### CODE REVIEW

**Comment by jiahaog** [comment]
Looks good! Would you like to have a go at implementing a menu bar option to clear the cache or shall I look into it myself?

**Comment by garymoon** [comment]
Yeah actually let me have a go, thanks :) Hit me up if you're getting impatient but I should have something (if no more than research) by tomorrow.

**Comment by jiahaog** [comment]
Sure! I believe it might be possible to get the session from the browser window, as per [the docs](https://github.com/atom/electron/blob/master/docs/api/session.md), and then possibly call one (or more) of the listed Session functions, like `clearCache` or `clearStorageData`. I've never worked with Session before so this might take a bit of testing.

**Comment by garymoon** [comment]
Definitely `clearStorageData`. I didn't see `clearCache`, that should definitely be included with `clearStorageData` and perhaps also as a standalone option? Also I think a reload of the view once it's done?

I'm thinking we want a confirmation dialog of some kind before proceeding with the full wipe, any thoughts on what we use for that?

**Comment by garymoon** [comment]
Please ignore me. I promise not to ask any more moronic questions before rtfm.

**Comment by garymoon** [comment]
I've added two options to the view menu (happy to move them to somewhere more appropriate). Clear data uses an electron dialog to prompt the user and reloads after a confirmation. Clear cache just reloads after clear without requesting confirmation. Let me know what you think.

There's a little bit of callback hell going on in the clear-app-data function, happy to take direction on how if you'd like to fix that up a bit.

**Comment by garymoon** [comment]
Really pissed off the linter. Fixed now.

**Comment by jiahaog** [comment]
Thanks! Is there a use case for the user when he wishes to clear the cache and clear the storage data separately? I was thinking we should simply have one button that clears both of them.

**Comment by jiahaog** [comment]
I also don't think it's necessary to check for a focused window, it should be alright to clear the data even if it is not focused.

Oh, it might be better to redirect the user back to the original targetUrl instead of reloading their current page after the data is cleared as well!

**Comment by garymoon** [comment]
All good points, though I feel the clear cache is useful (clear app data also does a clear cache currently). Granted I don't do it often, but sometimes a shift+refresh is the only way to clear up some stale data or some such. Happy to remove it if you disagree.

If i don't use the focusedWindow argument, can you point me to where I can get the session instance please?

**Comment by jiahaog** [comment]
Yeah, I think we can just clear both together in one command.

Could you try to pass the session instance into the menu from the `mainWindow.js` line 58 to add an additional parameter into the function to create a menu?

**Comment by garymoon** [comment]
I've removed the "clear cache' option.

I've added the `mainWindow` and `options` (for `.targetURL`) objects to the arguments for `createMenu` and refactored to remove other arguments that were methods of `mainWindow` (specifically `goBack`, `goForward` and `getURL`) and tested all three.

File: app/src/components/mainWindow/mainWindow.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ function createMainWindow(options, onAppQuit, setDockBadge) {
         mainWindow.webContents.send('change-zoom', currentZoom);
     };
 
-    createMenu(options.nativefierVersion, onAppQuit, mainWindow.webContents.goBack, mainWindow.webContents.goForward, onZoomIn, onZoomOut, mainWindow.webContents.getURL);
+    createMenu(options.nativefierVersion, onAppQuit, onZoomIn, onZoomOut, mainWindow, options);
```
Comment (jiahaog) [inline app/src/components/mainWindow/mainWindow.js]: What's the `options` here pass in for?

**Comment by jiahaog** [comment]
Everything looks good except for the options!

**Comment by garymoon** [comment]
Hehe. I expected you to ask, which is why I mentioned it in my last reply :P. `options` is passed in for `.targetUrl`. Happy to take direction on getting it from elsewhere.

**Comment by jiahaog** [comment]
Oops, I didn't notice that it was called in the function to reload the window. Everything looks good! I'll merge it in and then clean it up a little. Thank you so much for your help!!

**Comment by garymoon** [comment]
My pleasure, glad to be of help :)

**Pull request merged**